test/common/build/controllers.go

```diff
@@ -653,6 +649,10 @@ func RunBuildRunningPodDeleteTest(t testingT, clusterAdminClient *client.Client,
 	if newBuild.Status.Phase != buildapi.BuildPhaseError {
 		t.Fatalf("expected build status to be marked error, but was marked %s", newBuild.Status.Phase)
 	}
+
+	// allow time for the failed event to get created
+	time.Sleep(5 * time.Second)
```

Comment:
So the reason the delay is added here is because we immediately check whether an event was produced when we detect that a build was updated through the watch. Previously, in some code paths (like this one), we would emit the event before doing the update. Now we consistently emit the event after the update happened. Adding the wait here was my quick way of allowing the event to be produced after the build update happened. If you have an idea of how to better handle it, I'm open to suggestions :)